fs/checksum_calculator.go

```diff
@@ -14,11 +15,16 @@ import (
 // ChecksumCalculator can be used to calculate the SHA256 checksum of a given file or
 // directory. When given a directory, checksum calculation will be performed in
 // parallel.
-type ChecksumCalculator struct{}
+type ChecksumCalculator struct {
+	sumValue []byte
+	Hash     hash.Hash
```

Comment:
You probably don't mean to export this. If you did, then including a method on `ChecksumCalculator` that simply mirrors the `Write` method of the `hash.Hash` doesn't seem necessary.